### DISCUSSION TITLE
Can I use this without the API and just use it for Monolithic front-end and not headless?

### DISCUSSION THREAD

**Discussion by hamburger-fries** [prappo/wordpress-plugin-boilerplate#8] [Q&A] [answered]
Do I need to use Headless Front-End?
Or Can I use it in a monolithic format?

Or Would I do a hybrid approach and route the posts as a hybrid approach so its monolithic but also headless.

Basically I have a very large project and die to not wanting to code Ecommerce plugins and shipping plugins I want to use Wordpress monolithic for the time being.  

If i have to use the API to pass data to a Monolithic frontend how would I do that?

**Comment by Ajithpmdwaraka** [comment, marked as answer]
Hi bro, you can take a hybrid approach by using WordPress as a monolithic system for features like eCommerce and shipping plugins, while also leveraging its API to pass dynamic data to a custom front-end when needed. This way, your main site functions as a traditional WordPress-powered site, but you can use the REST API or GraphQL to send specific data to a decoupled front-end, such as a React or Vue.js app, for more flexibility in handling certain components. This allows you to have the convenience of a monolithic structure while still benefiting from headless architecture in specific areas, without fully committing to either approach.

**Reply by hamburger-fries** [reply]
Are you available for a 1 hour consult on a big project? Maybe I can pay you for the hour or invote you to work on my project? I run a high traffic 25+ year old project about Public Domain books.

Im on Discord: hamburger_fries

**Comment by Ajithpmdwaraka** [comment]
Hi brother, Thank you for the offer! I’m familiar with full stack web development and skilled in technologies such as JavaScript, TypeScript, React, Node.js, MongoDB, Express and many UI libraries. I’d love to learn more about your high-traffic project on public domain books and see how I can assist. Could you share the tech stack you’re using in your project? If it's within my expertise, I'd be happy to offer guidance and help where needed!

**Comment by prappo** [comment]
@forestcall You can access data either through the API or by passing your data directly as an array in the `/includes/Frontend/Frontend.php` file.

https://github.com/prappo/wordpress-plugin-boilerplate/blob/9260f624d5a7832191116f7177060430faa84fbc/includes/Frontend/Frontend.php#L104-L112

Access your data on the frontend using `wordpressPluginBoilerplate.apiUrl`, where the `wordpressPluginBoilerplate` object is globally available. You can change the object name as needed.

https://github.com/prappo/wordpress-plugin-boilerplate/blob/9260f624d5a7832191116f7177060430faa84fbc/includes/Frontend/Frontend.php#L30

Example:

https://github.com/prappo/wordpress-plugin-boilerplate/blob/9260f624d5a7832191116f7177060430faa84fbc/js/src/components/application-layout/LayoutOne.jsx#L58
